test/Common/DurableTaskEndToEndTests.cs

```diff
@@ -5758,12 +5758,17 @@ public async Task TestStoreInputsInOrchestrationHistory()
                    options))
             {
                 await host.StartAsync();
-                var client = await host.StartOrchestratorAsync(nameof(TestOrchestrations.SayHelloWithActivity), "foo", this.output);
-                await client.WaitForCompletionAsync(this.output);
-                var status = await client.InnerClient.GetStatusAsync(client.InstanceId, showHistory: true, showHistoryInput: true);
+                var client1 = await host.StartOrchestratorAsync(nameof(TestOrchestrations.SayHelloWithActivity), "foo", this.output);
+                var client2 = await host.StartOrchestratorAsync(nameof(TestOrchestrations.SayHelloWithActivity), null, this.output);

```

Comment:
You should make another orchestrator with two different activity functions. One function that takes a string input and one that doesn't take an input. In the scenario you currently have, null is being cast to a string and passed in the SayHello function. The reason I suggested this test was to make sure we return something for input when there isn't an input used at all.